src/Hl7.Fhir.ElementModel/ElementNode.cs

```diff
@@ -0,0 +1,220 @@
+﻿/* 
+ * Copyright (c) 2018, Firely (info@fire.ly) and contributors
+ * See the file CONTRIBUTORS for details.
+ * 
+ * This file is licensed under the BSD 3-Clause license
+ * available at https://raw.githubusercontent.com/ewoutkramer/fhir-net-api/master/LICENSE
+ */
+
+using Hl7.Fhir.Specification;
+using Hl7.Fhir.Utility;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace Hl7.Fhir.ElementModel
+{
+    public class DomNode<T> : IAnnotatable where T:DomNode<T>
```

Comment:
Do you want to have this class public?

Reply:
Unfortunately, you cannot make subclasses internal if a public class derives from them, so no, I don't want them public, but I have to.

Reply:
ah of course!